File: mhike-school-web/lib/report-quality.ts
import { apiPost } from "@/lib/api";

export type ReportQualityIssue = {
    type: string;
    message: string;
    suggestion: string | null;
};

export type ReportQualityResponse = {
    original_comment: string;
    corrected_comment: string;
    issues: ReportQualityIssue[];
};

export type ReportNotesGenerateResponse = {
    notes: string;
    generated_comment: string;
};

export async function checkReportComment(
    comment: string,
): Promise<ReportQualityResponse> {
    return apiPost<ReportQualityResponse>(
        "/report-quality/check-comment",
        {
            comment,
        },
    );
}

export async function generateReportFromNotes(
    notes: string,
    studentName?: string,
    subject?: string,
    yearGroup?: string,
): Promise<ReportNotesGenerateResponse> {
    return apiPost<ReportNotesGenerateResponse>(
        "/report-quality/generate-from-notes",
        {
            notes,
            student_name: studentName,
            subject,
            year_group: yearGroup,
        },
    );
}

File: app/schemas/report_quality.py
from __future__ import annotations

from pydantic import BaseModel, Field


class ReportQualityCheckRequest(BaseModel):
    comment: str = Field(
        min_length=1,
        description="The report comment to check and improve.",
    )


class ReportQualityIssue(BaseModel):
    type: str
    message: str
    suggestion: str | None = None


class ReportQualityCheckResponse(BaseModel):
    original_comment: str
    corrected_comment: str
    issues: list[ReportQualityIssue]


# ---------------------------------------------------------------------------
# Generate report from teacher notes
# ---------------------------------------------------------------------------


class ReportNotesGenerateRequest(BaseModel):
    notes: str = Field(
        min_length=1,
        description="Teacher notes or bullet points.",
    )
    student_name: str | None = None
    subject: str | None = None
    year_group: str | None = None


class ReportNotesGenerateResponse(BaseModel):
    notes: str
    generated_comment: str
